coordinator/gscoordinator/kubernetes_launcher.py

```diff
@@ -192,10 +198,50 @@ def __init__(
         self._mars_worker_cpu = mars_worker_cpu
         self._mars_worker_mem = mars_worker_mem
 
+        # check the validity of mode
+        self._mode = mode
+        if self._mode not in ["eager", "lazy"]:
+            logger.error(
+                "Invalid mode: %s, mode must be one of eager or lazy", self._mode
+            )
+
+        self._vineyard_pod_name_list = []
+
+        if self.vineyard_deployment_exists():
+            self._deploy_vineyard_deployment_if_not_exist()
+            # check the if the vineyard deployment is ready again
+            if not self._check_if_vineyard_deployment_exist():
+                # if not ready, then set the vineyard deployment to None
+                self._vineyard_deployment = None
+
+        # if the vineyard deployment is not set and use the eager mode,
+        # which means deploy the engine as a single pod and there is no
+        # external vineyard deployment. The vineyard objects are not
+        # shared between the engine pods, so raise an error here.
+        if self._mode == "lazy" and self._vineyard_deployment is None:
+            raise ValueError("If the mode is lazy, the vineyard deployment must be set")
```

Comment:
Rephrase the error message to:  Lazy mode is only possible with a vineyard deployment, please add a vineyard deployment name by `vineyard_deployment='vy-deploy'`
Proceeding as eager mode.

And set the mode to eager, don't raise error, continuing with eager mode.